### PR TITLE
CANBUS driver for native_posix board

### PR DESCRIPTION
**Pull request by jukkar**
Edit: I am now able to receive and send CAN packets in Zephyr. The Linux vcan virtual interface is the peer in this case. My earlier problems were because filters were not set properly. Currently this only supports Socket-CAN interface in Zephyr, we can add support for Zephyr native API later.

In Linux one can use cansend to send CAN packets like this:
```
./cansend zcan 001#11.2233.44556677.88
```

To receive CAN packets do this:
```
./candump zcan,1:1
```

The candump and cansend are part of the can-utils package https://github.com/linux-can/can-utils
You also need the can-setup.sh script from Zephyr net-tools repo, I will send new version for it shortly.

This PR contains two extra commits (padding and const ones) that I already sent separately, just ignore them.

### CODE REVIEW

File: drivers/can/Kconfig.native_posix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	range 1 56
+	help
+	  Defines the array size of the callback/msgq pointers.
+	  Must be at least the size of concurrent reads.
+
+config CAN_NATIVE_POSIX_INTERFACE_NAME
+	string "CANBUS interface name in Linux side"
+	default "zcan"
+	help
+	  This option sets the CANBUS network interface name in host system.
```
Comment (aescolar) [inline drivers/can/Kconfig.native_posix]: minor: name in +the+ host system.

File: drivers/can/canbus_native_posix_adapt.c [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/select.h>
+#include <net/if.h>
+#include "posix_trace.h"
+
+#ifdef __linux
+#include <linux/can.h>
+#endif
```
Comment (aescolar) [inline drivers/can/canbus_native_posix_adapt.c]: would a #warning or #error under an `#else` make sense here?

File: drivers/can/Kconfig.native_posix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+config CAN_MAX_FILTER
+	int "Maximum number of concurrent active filters"
+	depends on CAN_NATIVE_POSIX
+	default 32
+	range 1 56
+	help
+	  Defines the array size of the callback/msgq pointers.
+	  Must be at least the size of concurrent reads.
+
+config CAN_NATIVE_POSIX_INTERFACE_NAME
```
Comment (aescolar) [inline drivers/can/Kconfig.native_posix]: Sidecomment: If you want you can add a command line option to set it, similarly to:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/bluetooth/hci/userchan.c#L239

**Review by aescolar — COMMENTED** [review]
would you like to add a paragraph or two of documentation?

Otherwise just a couple of very minor comments after a quick lock

**Comment by jukkar** [comment]
> would you like to add a paragraph or two of documentation?

Yes, I will create documentation for this. I just didn't had time to create one today. This is probably not going to 1.14 anyway.

File: drivers/can/Kconfig.native_posix (hunk 3 above)
Comment (jukkar) [reply drivers/can/Kconfig.native_posix]: Thanks for the hint. That looks like a way to go.

**Review by pfalcon — APPROVED** [review]
Thanks for looking into this. Wouldn't have a chance to play with this soon, so just approving.

**Comment by jukkar** [comment]
Rebased against latest master.

**Comment by jukkar** [comment]
This is not ready yet as I need to still solve the issues @aescolar mentioned.

**Comment by aescolar** [comment]
@jukkar Note that this PR is getting stale

**Comment by jukkar** [comment]
It is stale already and would need more TLC. Let's see if I have time to work more with it.

**Comment by jukkar** [comment]
Storing this to https://github.com/jukkar/zephyr/pull/21 in order to cleanup the upstream list of PRs.

**Comment by pllee4-weston** [comment]
hi @jukkar can this issue be reopened? We have rebased your previous work and hope it can go upstream.

**Comment by jukkar** [comment]
@pllee4-weston Please create a new pull request with your changes, it is not possible to reopen this PR as the underlying branch has been deleted from github.

**Comment by pllee4-weston** [comment]
@jukkar Alright thanks for the information.

**Pull request closed without merging**